src/shapes.ts

```diff
@@ -1,34 +1,52 @@
-import { Scene } from "hovl/scene";
 import { Vector } from "hovl/vector";
 
+interface Trs {
+  translation: Vector;
+  rotation: number;
+  scale: Vector | number;
+}
+
 export abstract class Shape {
-  public pos: Vector;
   public color: string;
 
-  constructor(x: number, y: number, color: string) {
-    this.pos = new Vector(x, y);
+  public translation: Vector;
+  public rotation: number;
+  public scale: Vector;
+
+  constructor(color: string, trs: Partial<Trs>) {
     this.color = color;
+    this.translation = trs.translation || new Vector(0, 0);
```

Comment:
Maybe it's a good idea to use `trs.translation.copy()` to avoid weird side effects if the trs object is modified from outside this class. That would apply to rotation and scale as well.